src/util/schnorr.rs

```diff
@@ -38,7 +39,10 @@ pub trait TapTweak {
     ///  * H is the hash function
     ///  * c is the commitment data
     ///  * G is the generator point
-    fn tap_tweak<C: Verification>(&self, secp: &Secp256k1<C>, merkle_root: Option<&TapBranchHash>) -> TweakedPublicKey;
+    ///
+    /// # Returns
+    /// The tweaked key and its parity.
+    fn tap_tweak<C: Verification>(&self, secp: &Secp256k1<C>, merkle_root: Option<&TapBranchHash>) -> (TweakedPublicKey, bool);
```

Comment:
`bool` is not very clear value for parity. Which is which? I believe it'd be much better to have:

```rust
enum KeyParity {
    Even,
    Odd,
}
```

However, I think this can go to a separate PR given that `parity: bool` was used already.

Reply:
I'll do the PR, thanks.

Reply:
lol, I started doing this over in `rust-secp256k1` and I do not know which is which. Is `true` == `Even` or `Odd`?

Reply:
https://github.com/rust-bitcoin/rust-secp256k1/pull/344